consensus/event_loop.go

```diff
@@ -165,41 +166,35 @@ func (fc *FastHotStuffEventLoop) ProcessTipBlock(tip BlockWithValidators, safeBl
 	return nil
 }
 
-// setSafeBlocks is a helper function that validates the provided blocks, validator sets, and stores them.
+// storeBlocks is a helper function that validates the provided blocks, validator lists, and stores them.
 // It must be called while holding the event loop's lock.
-func (fc *FastHotStuffEventLoop) storeBlocks(tip BlockWithValidators, safeBlocks []BlockWithValidators) error {
-	// Do a basic integrity check on the tip block and validator set
-	if !isProperlyFormedBlock(tip.Block) || !isProperlyFormedValidatorSet(tip.Validators) {
-		return errors.New("Invalid tip block or validator set")
+func (fc *FastHotStuffEventLoop) storeBlocks(tip BlockWithValidatorList, safeBlocks []BlockWithValidatorList) error {
```

Comment:
Is this tip the uncommitted tip? should we make that explicit?

Reply:
It could be the committed or uncommitted tip (depending on the state of the chain). Just whatever the the last block in the chain is, this is it.